src/libexpr/primops.cc

```diff
@@ -3951,6 +3951,27 @@ static RegisterPrimOp primop_div({
     .fun = prim_div,
 });
 
+static void prim_mod(EvalState & state, const PosIdx pos, Value * * args, Value & v)
+{
+    auto i2 = state.forceInt(*args[1], pos, "while evaluating the second operand of the modulo");
+
+    if (i2.value == 0)
+        state.error<EvalError>("modulo by zero").atPos(pos).debugThrow();
+
+    auto i1 = state.forceInt(*args[0], pos, "while evaluating the first operand of the modulo");
+
+    v.mkInt(i1.value % i2.value);
+}
+
+static RegisterPrimOp primop_mod({
+    .name = "__mod",
+    .args = {"e1", "e2"},
+    .doc = R"(
+      Return the remainder of the division of the numbers *e1* and *e2*.
```

Comment:
Based on your response:

```suggestion
      Return the remainder of the division of the numbers *e1* and *e2*.
      
      The outcome is **negative** when the remainder is non-zero, and exactly one of *e1* or *e2* evaluates to a negative number, so that the following identity holds:
      ```nix
      a == b * (a / b) + (a % b)
      ```
```

Reply:
I think that misses the point - what do you think about flipping your suggestion on its head?

start out with that identity because that's the main point - modulo defined by the behaviour of division - then later state what you stated in the preceding sentence so that it's not just dry math and you have some rule that might explain it better

also probably good to mention that the identity is all integer operations

Reply:
@roberth could you check the current doc?

Reply:
Great!

Reply:
Maybe I'm missing something, but looking at the four examples in <https://github.com/NixOS/nix/pull/12617#issuecomment-2708809929>, shouldn't it be: "the outcome is negative when the remainder is non-zero and e1 evaluates to a negative number"?